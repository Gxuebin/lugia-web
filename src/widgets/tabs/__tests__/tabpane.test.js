/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Tabpane from '../tabpane';

Enzyme.configure({ adapter: new Adapter() });

describe('tabpaneDemo', () => {
  it('Component JSON', () => {
    const renders = renderer.create(<Tabpane />);
    expect(renders.toJSON()).toMatchSnapshot();
  });
  const getCmp = (target: any): Object => {
    return target
      .children()
      .at(0)
      .instance();
  };
  it('props activityKey', () => {
    const target = mount(<Tabpane activityKey={'2'} />);
    expect(getCmp(target).props.activityKey).toBe('2');
  });

  it('props isSelect', () => {
    const target = mount(<Tabpane isSelect={true} />);
    expect(getCmp(target).props.isSelect).toBe(true);
  });
  function testTitle(title: string, expTitle: string) {
    it('props title', () => {
      const target = mount(<Tabpane title={title} />);
      expect(
        target
          .find('div')
          .at(1)
          .text()
      ).toBe(expTitle);
    });
  }
  testTitle('1111', '1111');
  testTitle('1234', '1234');
  testTitle('tabs', 'tabs');
  testTitle('', '');
  it('props disabled true  ', () => {
    const target = mount(<Tabpane activityKey={'2'} disabled={true} />);
    expect(getCmp(target).props.disabled).toBe(true);
  });
  it('props disabled false', () => {
    const target = mount(<Tabpane activityKey={'2'} disabled={false} />);
    expect(getCmp(target).props.disabled).toBe(false);
  });
});
