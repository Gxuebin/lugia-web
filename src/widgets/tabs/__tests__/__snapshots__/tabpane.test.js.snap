// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tabpaneDemo Component JSON 1`] = `
.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  font-size: 1.2rem;
}

.c0 {
  position: relative;
  cursor: pointer;
  white-space: nowrap;
  cursor: pointer;
  display: inline-block;
  line-height: 2.8333333333333335em;
  height: 2.8333333333333335em;
  padding: 0 0.8333333333333334em 0 1.6666666666666667em;
  box-shadow: 0 0 0.5em none;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  background: none;
  z-index: 5;
  box-sizing: border-box;
}

.c0:hover > div {
  color: #684fff;
}

.c0:hover > span > i:first-child {
  color: none;
}

.c1 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  text-align: left;
  padding: 0 0em 0 0em;
  height: 2.8333333333333335em;
  line-height: 2.8333333333333335em;
}

.c1:focus {
  color: #684fff;
}

.c2 {
  -webkit-transition: all 0.3s linear 0.1s;
  transition: all 0.3s linear 0.1s;
  z-index: 2;
  display: inline-block;
  opacity: 1;
}

.c3 {
  font-size: 1rem;
}

.c3:hover {
  color: #999999;
}

<div
  className="c0"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="c1"
  />
  <span
    className="c2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <i
      className="lugia-icon-reminder_close c3 c4"
    />
  </span>
</div>
`;
